src/main/java/uk/gov/cshr/civilservant/controller/CivilServantController.java

```diff
@@ -105,6 +114,83 @@ public CivilServantController(LineManagerService lineManagerService, CivilServan
         return ResponseEntity.unprocessableEntity().build();
     }
 
+    @GetMapping("/org")
+    public ResponseEntity<OrgCodeDTO> getOrgCodeForCivilServant(@RequestParam(value = "uid") String uid) {
+        log.debug("Getting civil servant org details for user with uid " + uid);
+
+        Optional<CivilServant> civilServant = civilServantRepository.findByIdentity(uid);
+
+        if (civilServant.isPresent()) {
+            return civilServantResourceFactory.getCivilServantOrganisationalUnitCode(civilServant.get())
+                    .map(orgCodeDTO -> ResponseEntity.ok(orgCodeDTO))
+                    .orElse(buildNotFound(uid));
+        } else {
+            return buildNotFound(uid);
```

Comment:
I'd prefer not to refactor the return of the `ResponseEntity` into a method, as it should be clear what the return of the controller method is.

Reply:
Updated